src/test/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSourceUnitTest.java

```diff
@@ -98,6 +102,34 @@ public void testPartitionSizing(){
         Assert.assertEquals(smallPartitions.partitions().size(), 2);
     }
 
+    @Test
+    public void testReadFromFileAndHDFS() throws IOException {
+        final File bam = new File( getToolTestDataDir(), "reads.bam");
+        final File bai = new File( getToolTestDataDir(), "reads.bai");
+        MiniDFSCluster cluster = null;
+        try {
+            cluster = new MiniDFSCluster.Builder(new Configuration()).build();
+            final Path workingDirectory = cluster.getFileSystem().getWorkingDirectory();
+            final Path bamPath = new Path(workingDirectory,"flag_stat_hdfs.bam");
+            final Path baiPath = new Path(workingDirectory, "flag_stat_hdfs.bai");
+            cluster.getFileSystem().copyFromLocalFile(new Path(bam.toURI()), bamPath);
+            cluster.getFileSystem().copyFromLocalFile(new Path(bai.toURI()), baiPath);
```

Comment:
that's a good question... I was assuming yes... but on further inspection I don't know